java-lib/src/main/java/com/wavefront/ingester/SourceTagDecoder.java

```diff
@@ -0,0 +1,36 @@
+package com.wavefront.ingester;
+
+import java.util.List;
+
+import sunnylabs.report.ReportSourceTag;
+
+/**
+ * This class is used to decode the source tags sent by the clients.
+ *
+ * [@SourceTag action=save source=source sourceTag1 sourceTag2]
+ * [@SourceDescription action=save source=source description=Description]
+ *
+ * @author Suranjan Pramanik (suranjan@wavefront.com).
+ */
+public class SourceTagDecoder {
+
+  private static final String SOURCE_TAG = "@SourceTag";
+  private static final String SOURCE_DESCRIPTION = "@SourceDescription";
+
+  private static final SourceTagIngesterFormatter FORMAT = (SourceTagIngesterFormatter)
+      SourceTagIngesterFormatter
```

Comment:
ditto re: generics, also we wouldn't typically line break with just a class without a method call (remove new line on line 21)